net.c

```diff
@@ -95,10 +95,10 @@ send_remote_command(int fd, const char* fmt, ...)
 	strcat(cmd, "\r\n");
 	len = strlen(cmd);
 
-	if (((config.features & SECURETRANSFER) != 0) &&
-	    ((config.features & NOSSL) == 0)) {
-		while ((s = SSL_write(config.ssl, (const char*)cmd, len)) <= 0) {
-			s = SSL_get_error(config.ssl, s);
+	if (is_configuration_setting_enabled(CONF_SECURETRANSFER) == true &&
```

Comment:
with `ssl_active`, you can drop checking the config setting

Reply:
I'll reply here for all the comments on ssl_state or no_ssl_flag:
As far as I understand the previous code, the NOSSL flag was just meant as a temporary "setting" for disabling SSL while establishing a STARTTLS connection.
So, this flag does mean, that ssl is en- oder disabled in the config, but rather if it's (temporarily) disabled despite it being enabled in the config.

As to its name: Maybe lock_ssl or so would be more appropriate in this case (if I've understood your code correctly).

Reply:
I think the original code was a bad hack in the first place.  It probably makes more sense to have a flag (along with `ssl_state`), which says "and now also use SSL IO functions", rather than saying "oh don't use SSL IO functions".

Reply:
Well but that is exactly what this flag seems to be, it says: "Yes, SSL may be enabled or not, but we won't use it anyway" It's overriding the settings as far as I can tell it.
Imo ssl_active isn't a good name in that case, I'd suggest something like "ssl_override" but of course you have the final say on that.